pkg/kubelet/eviction/helpers.go

```diff
@@ -669,35 +661,87 @@ func memory(stats statsFunc) cmpFunc {
 
 		// adjust p1, p2 usage relative to the request (if any)
 		p1Memory := p1Usage[v1.ResourceMemory]
-		p1Request := podMemoryRequest(p1)
+		p1Request := podRequest(p1, v1.ResourceMemory)
 		p1Memory.Sub(p1Request)
 
 		p2Memory := p2Usage[v1.ResourceMemory]
-		p2Request := podMemoryRequest(p2)
+		p2Request := podRequest(p2, v1.ResourceMemory)
 		p2Memory.Sub(p2Request)
 
 		// if p2 is using more than p1, we want p2 first
 		return p2Memory.Cmp(p1Memory)
 	}
 }
 
-// podMemoryRequest returns the total memory request of a pod which is the
+// podRequest returns the total memory request of a pod which is the
 // max(sum of init container requests, sum of container requests)
-func podMemoryRequest(pod *v1.Pod) resource.Quantity {
+func podRequest(pod *v1.Pod, resourceName v1.ResourceName) resource.Quantity {
 	containerValue := resource.Quantity{Format: resource.BinarySI}
+	if resourceName == resourceDisk && !utilfeature.DefaultFeatureGate.Enabled(features.LocalStorageCapacityIsolation) {
+		return containerValue
+	}
 	for i := range pod.Spec.Containers {
-		containerValue.Add(*pod.Spec.Containers[i].Resources.Requests.Memory())
+		switch resourceName {
+		case v1.ResourceMemory:
+			containerValue.Add(*pod.Spec.Containers[i].Resources.Requests.Memory())
+		case resourceDisk:
```

Comment:
Do you plan to eventually add v1.ResourceDisk?

Reply:
I intend to move the entire eviction logic to use v1.EphemeralStorage in a seperate PR